### DISCUSSION TITLE
Converting a JPG to PNG and Making the Background Transparent

### DISCUSSION THREAD

**Discussion by trythisone35** [ImageMagick/ImageMagick#6605] [Help] [answered]
### ImageMagick version

7.1.1-15-Q16-HDRI-x64

### Operating system, version and so on

Windows 10

### Description

I found some suggestions on solving this by using the following:

convert initial.jpg -fill none -fuzz 20% -draw 'matte 0,0 floodfill' result.png

When I try this I get an error "non-conforming drawing primitive definition 'matte' @error/daw.c/RenderMVGContent/4507

Using only the fuzz gets me close but some of the actual image become transparent and I'd like to avoid that if possible and saw that using floodfill might help.   Is my syntax incorrect?

![2P134412](https://github.com/ImageMagick/ImageMagick/assets/79946966/1776d822-fba7-4988-92c7-4449b7316a66)

**Comment by fmw42** [comment]
Your shirt color is too close to the background color to do this processing only as image processing.  This kind of background removal would need AI/Deep Learning with lots of training. Or use one of the web sites such as http://remove.bg.

If you have images where the shirt color is a good contrast from the background color, then something like what you are trying to do is possible.

However as your background is pure white, then this is possible in Imagemagick as

`magick 264470857-1776d822-fba7-4988-92c7-4449b7316a66.jpg -fill none -draw "alpha 0,0, floodfill" result.png`

![result](https://github.com/ImageMagick/ImageMagick/assets/13458218/4846452e-7267-4cde-a532-cfca8d24e88a)

**Download the file to actually see where it is transparent.**

**Reply by trythisone35** [reply]
Thank you for the quick response and help.  That command line is working.   I will experiment some more.  Again, I appreciate your help.

**Reply by trythisone35** [reply]
I have one additional question.  I noticed in testing some other images that the edges have some stray pixels along the edges.  Is it possible to clean those up?

I provided another example showcasing it.  


![whitebody](https://github.com/ImageMagick/ImageMagick/assets/79946966/7155d810-6f47-4aa5-bc87-934594c8f769)
![straypixels](https://github.com/ImageMagick/ImageMagick/assets/79946966/9b506335-b9dc-4703-bcee-5b0c5875c703)

**Comment by fmw42** [comment, marked as answer]
One way is to use morphology to clean up small spots.

![mask](https://github.com/ImageMagick/ImageMagick/assets/13458218/224efc23-1d1c-469c-931d-bb35a89bc742)

`magick mask.png -morphology open diamond:1 -morphology close diamond:1 mask_proc.png`

![mask_proc](https://github.com/ImageMagick/ImageMagick/assets/13458218/2416f814-e584-4c0e-b3e8-8d2b9251c603)


You can also smooth it by using -blur

`magick mask.png -morphology open diamond:1 -morphology close diamond:1 -blur 0x2 -level 0x50% mask_proc_2.png`

![mask_proc_2](https://github.com/ImageMagick/ImageMagick/assets/13458218/9a477e35-25c2-4dd1-b9dd-054c385226d9)

**Reply by fmw42** [reply]
P.S. The spots come from the JPG lossy compression of your input image when thresholded.  It is always best to use and image format such as PNG or TIFF that does not use lossy compression, if possible.

**Reply by trythisone35** [reply]
Interesting.  Our source file is a JPG so that's what I have to work with.
